pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/RecordTransformer.java

```diff
@@ -18,15 +18,79 @@
  */
 package org.apache.pinot.segment.local.recordtransformer;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
 import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
 import javax.annotation.Nullable;
+import org.apache.pinot.plugin.record.enricher.RecordEnricherRegistry;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.ingestion.EnrichmentConfig;
+import org.apache.pinot.spi.config.table.ingestion.IngestionConfig;
 import org.apache.pinot.spi.data.readers.GenericRow;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 
 /**
  * The record transformer which takes a {@link GenericRow} and transform it based on some custom rules.
  */
 public interface RecordTransformer extends Serializable {
+  final boolean GROOVY_DISABLED = false;
```

Comment:
Suggest keeping the interface simple. We need to add `getInputColumns()` from `RecordEnricher`, other method seems unnecessary. Take a look at `CompositeTransformer` for record transformer handling